### DISCUSSION TITLE
Avalonia 11 preview 7 How does ItemsControl delete items?

### DISCUSSION THREAD

**Discussion by xinghaiyege** [AvaloniaUI/Avalonia#11126] [Q&A] [answered]
This is the previously valid code.

```c#
ItemsControl .ItemsPanelRoot.Children.Remove(menu.Parent.Parent as WrapPanel);
```

**Comment by xinghaiyege** [comment, marked as answer]
Instead of deleting the `Control` in the `DataTemplate`, delete the `ContentPresenter`, the parent control of this `Control`.

```c#
using Avalonia.Controls.Presenters;

ItemsControl .ItemsPanelRoot.Children.Remove(menu.Parent.Parent.Parent as ContentPresenter);
```

**Comment by grokys** [comment]
You shouldn't be manipulating the items panel manually, instead you should remove the items from the `Items` or `ItemsSource` collection.

**Reply by xinghaiyege** [reply]
I think this method will reduce the performance efficiency of the application, so I manually deleted the child controls.
